lua/gitblame/init.lua

```diff
@@ -563,18 +564,62 @@ local function clear_all_extmarks()
     end
 end
 
-local function disable()
-    if vim.g.gitblame_enabled == 0 then
+local function set_autocmds()
+    local autocmd = vim.api.nvim_create_autocmd
+    local group = vim.api.nvim_create_augroup("gitblame", { clear = true })
+
+    autocmd("CursorMoved", { callback = schedule_show_info_display, group = group })
+    autocmd("CursorMovedI", { callback = clear_virtual_text, group = group })
+    autocmd("InsertEnter", { callback = clear_virtual_text, group = group })
+    autocmd("TextChanged", { callback = handle_text_changed, group = group })
+    autocmd("InsertLeave", { callback = handle_insert_leave, group = group })
+    autocmd("BufEnter", { callback = handle_buf_enter, group = group })
+    autocmd("BufDelete", { callback = cleanup_file_data, group = group })
+end
+
+M.disable = function(force)
+    if not vim.g.gitblame_enabled and not force then
         return
     end
 
-    vim.g.gitblame_enabled = 0
-
+    vim.g.gitblame_enabled = false
+    pcall(vim.api.nvim_del_augroup_by_name, "gitblame")
```

Comment:
```suggestion
    vim.api.nvim_del_augroup_by_name("gitblame")
```

Reply:
`vim.api.nvim_del_augroup_by_name("gitblame")` fails when group does not exists. This happens when enabled is false. So using `pcall()`

Reply:
Ah, gotcha!